IABTechLab/uid2-self-serve-portal
# Change URL scheme to include participant ID

## What changed
- Use the new scheme of `/participant/:participantId/<page_name>` rather than `/dashboard/<page_name>`. Including giving the home page a well-deserved path name. Added a re-direct from legacy routes to the new root.
- Added a wrapper function for `makeLoader` which makes a loader that is enriched with the `participantId` from the URL. This allows all the loader functions to be called with the `participantId` - so we can start loading data before the React components are aware of the URL.
- Add a new landing component that is hit when the user logs in. This component loads the first participant of the user then navigates to the appropriate home URL for that participant.
- Get rid of usage of `<SnailTrail>` in the dashboard - I don't think it was actually doing anything.
- A bunch of other changes to navigation. Trying to use relative paths were possible (e.g. `RotateApiKeyCard.tsx`), but otherwise need to use the  `useParams()` hook to populate the `participantId` in the path properly (e.g. `PortalHeader.tsx`)

## Manual testing
1. Try do everything you usually would in the portal, including navigation, creating keys, etc.
2. You'll notice a bunch of the API calls have the `participantId` specified in the request URL (e.g. `http://localhost:3000/api/participants/8/apiKeys`), rather than relying on "current": `http://localhost:3000/api/participants/current/apiKeys`.
3. Hit a legacy URL e.g. `http://localhost:3000/dashboard/apiKeys` and either be routed to the new home route (if authenticated) or the login page (if unauthenticated)

## Future work
- Remove support for "current" in the web services - i.e. make `participantId` mandatory where it is often currently optional
- Remove `<SnailTrail>` component
